WebClient/Scripts/temperatureControlView.js

```diff
@@ -0,0 +1,477 @@
+        this.centerColor = "#CCD1D1";
+
+        //min and max temperature in K
+        this.minTemperature = new UnitConverter.ConvNum('Temperature', 253.15);
+        this.maxTemperature = new UnitConverter.ConvNum('Temperature', 323.15);
+
+        this.TemperatureUnits = this.minTemperature.converter.units;
+
+        //Universal Scale
+        this.universalScale = d3.scale.linear()
+            .domain([this.minTemperature.GetDisplayValue(), this.maxTemperature.GetDisplayValue()])
+            .range([this.tubeHeight - this.tubeBulbPixelOverlap, 0]);
+
+        /*
+        //Todo: D3 V4 code for the scale
+        this.universalScale = d3.scaleLinear()
+            .domain([this.minTemperature.GetDisplayValue(), this.maxTemperature.GetDisplayValue()])
+            .range([this.tubeHeight - this.tubeBulbPixelOverlap, 0]);
+        */
+
+        //selected unit variable - default set to C
+        this.selectedUnit = this.TemperatureUnits[1];//TemperatureUnit.C;
+
+        //setting up the temperature model and passing in the default values
+        this.temperatureModel = new TemperatureModel({
+            valueK: 273.15,
+            unit: this.selectedUnit,
+        });
+
+        //connecting the model event
+        this.temperatureModel.on('value', this.modelValueChanged.bind(this));
+        this.temperatureModel.on('temperatureUnit', this.modelValueChanged.bind(this));
+
+        //mouse click Y-position for the mercury level
+        this.mouseClickY;
+
+        //this flag becomes true when left click is detected (left click - edit temperature; right click - fetch live temperature)
+        this.editTemperatureFlag = false;
+
+        //-----------------Creating and adding components------------------------
+        
+        this._container = this.parentContainer;
+
+        this._temperatureDiv = L.DomUtil.create('div', "temperatureDiv");
+        this._container.appendChild(this._temperatureDiv);
+
+        //disabling propagation
+        L.DomEvent.disableClickPropagation(this._container);
+        L.DomEvent.disableScrollPropagation(this._container);
+
+        this.temperatureDiv = d3.select(this._temperatureDiv).style("width", this.width + "px").style("height", this.height + "px");
+
+        this.unitControlParentDivInitialization();
+
+        this.temperatureTextboxInitialization();
+
+        this.svgImageInitialization();
+
+        //definition of the mercury circle color gradient component
+        this.defs = this.svg.append("defs");
+        this.mercuryBulbGradientInitialization("mercuryGradient_red", this.mercuryColor_red);
+        this.mercuryBulbGradientInitialization("mercuryGradient_blue", this.mercuryColor_blue);
+
+        this.createThermometerStructure();
+        this.createMercuryBar();
+        this.axisInitialization();
+
+        //setting the default startup value for the temperature control
+        this.temperatureModel.value = this.temperatureModel.value;
```

Comment:
Replace with an explicit callback call.
`this.modelValueChanged(this.temperatureModel.value)`

Reply:
Done